### DISCUSSION TITLE
[March 2024] zkSync developer update

### DISCUSSION THREAD

**Discussion by uF4No** [zkSync-Community-Hub/zksync-developers#437] [Announcements]
Greetings devs! Welcome to the zkSync developer update where we showcase different updates about the protocol, developer tools, integrations, tutorials, developer-oriented content and more.

Here are the updates for the last month:

### 👷 👷‍♀️Protocol upgrades

- **Dencun upgrade:** the Ethereum Dencun upgrade was activated on March 13th introducing proto-danksharding (aka [EIP4844](https://www.eip4844.com/)) with blobs aimed to be used by L2s for data availability. If you want to learn more about the technical updates we did on zkSync Era to support EIP4844, check out [this presentation by protocol engineer Zach Kolodny](https://www.youtube.com/watch?v=O7VERaQCxQ4&t=7319s). zkSync Era was the first L2 to make use of the blobs and transactions fees have been consistently reduced by up to 10x.

<img width="838" alt="Screenshot 2024-04-03 at 12 18 05" src="https://github.com/zkSync-Community-Hub/zksync-developers/assets/1957117/6a619a84-f0d0-43de-a6ce-cfb2636699ec">

- **Goerli is dead, long live Sepolia:** as we [previously announced](https://github.com/zkSync-Community-Hub/zksync-developers/discussions/228), the zkSync Goerli testnet has been deprecated. From now on, all projects must use the zkSync Sepolia testnet. You can find the network details in [this section of the docs](https://docs.zksync.io/build/quick-start/add-zksync-to-metamask.html#sepolia-testnet-network-info).

### $ Native USDC on zkSync Era

Native USDC issued by Circle is available on zkSync Era. The current version USDC used in most DeFi apps on zkSync Era is bridged from Ethereum and, to avoid confusion, will be renamed to USDC.e. The list of tokens and its addresses will be:

**Native USDC issued by Circle**

- Token Symbol: USDC
- Mainnet Token Address: [0x1d17CBcF0D6D143135aE902365D2E5e2A16538D4](https://explorer.zksync.io/address/0x1d17CBcF0D6D143135aE902365D2E5e2A16538D4)
- Testnet USDC Address: [0xAe045DE5638162fa134807Cb558E15A3F5A7F853](https://sepolia.explorer.zksync.io/address/0xAe045DE5638162fa134807Cb558E15A3F5A7F853)

**USDC bridged from Ethereum**

- Token Symbol: USDC.e 
- Token Address: [0x3355df6D4c9C3035724Fd0e3914dE96A5a83aaf4](https://explorer.zksync.io/address/0x3355df6D4c9C3035724Fd0e3914dE96A5a83aaf4)

Find more info in [this blog post from Circle](https://www.circle.com/blog/what-you-need-to-know-native-usdc-on-zksync).

### 🛠️ Developer tools

- **era-test-node:** alpha 19 was released with fixes to adjust to the new fee model. Find the [release notes here](https://github.com/matter-labs/era-test-node/releases/tag/v0.1.0-alpha.19).
- **hardhat-zksync-deploy:** new version of the plugin that simplifies deployment of smart contracts in Hardhat, which includes the introduction of `hre.deployer` and caching. Read the [[full docs here](https://docs.zksync.io/build/tooling/hardhat/hardhat-zksync-deploy.html#caching-mechanism)](https://docs.zksync.io/build/tooling/hardhat/hardhat-zksync-deploy.html#caching-mechanism).
- **viem:** release of the `deployContract` action to deploy contracts on zkSync using viem. Find more info in [the viem/zksync documentation.](https://viem.sh/zksync/actions/deployContract)
- **Signature-based paymaster:** a new type of paymaster added to the [paymaster-examples repository](https://github.com/matter-labs/paymaster-examples). This paymaster will cover the transaction fees for a user until a signature expires. Thanks @hoshiyari420 for the contribution.
- **USDC faucet:** Circle has released a [USDC faucet](https://faucet.circle.com/) with support for zkSync Era Sepolia testnet.

### 📖 Workshops & tutorials

- **Magic and Zyfi:** [Magic](https://magic.link/) is a wallet-as-a-service that allows developers to create invisible wallets using common authentication providers (like Google or Apple). In combination with [Zyfi’s paymaster as a service](https://www.zyfi.org/), you can create apps that abstract all blockchain complexity with gasless transactions. Read the [full tutorial in the magic documentation](https://magic.link/docs/wallets/integrations/zksync-zyfi-integration).
- **Chainlink workshop:** we co-hosted a workshop with Chainlink covering how to deploy a smart contract on zkSync and how to interact with [Chainlink’s price feeds on zkSync Era](https://docs.chain.link/data-feeds/price-feeds/addresses?network=zksync&page=1#zksync-era-mainnet). Find [the recording here](https://www.youtube.com/watch?v=rGdt-CNa3W4).
- **Native account abstraction explained like I’m 5:** another community contribution by [Mohiii](https://twitter.com/MohiiiEth) with an awesome online book about native account abstraction for everyone. [Check it out here](https://www.figma.com/proto/tz5DB8NzlJ9K2aRtErUKBl/Zeek?page-id=195%3A126&type=design&node-id=305-22452&viewport=12247%2C-300%2C0.33&t=HjLINvjcbvwu3vib-1&scaling=contain&starting-point-node-id=305%3A22452).

### 🎂 zkSync Era anniversary

March 24th was the one year anniversary of zkSync Era on mainnet. During this year the network has processed:

- 328 million transactions
- 6.1 million accounts
- 1.6 million smart contracts deployed

You can find more details in [the following Dune dashboard](https://dune.com/matter_labs/zksync-era-overview) and [l2beat](https://l2beat.com/scaling/projects/zksync-era).

### 🔗 Stay connected

- Follow [@zkSyncDevs](https://twitter.com/zkSyncDevs) on X
- Join the #dev-general channel in our Discord

Thanks for being part of the zkSync developer community. Your support and contributions are what make this project thrive.

Keep buidling! 👩‍💻🧑🏻‍💻

**Comment by 0tayfun7** [comment]
Congratulations. Good work.

**Comment by Ozkulo17** [comment]
Good work.

**Comment by nickolay-avasiloy** [comment]
congrats 🥳
